### PR TITLE
Add type checking to remaining actions / reducers

### PR DESCRIPTION
**Pull request by wolverineks**

### CODE REVIEW

File: src/modules/UI/Settings/reducer.js [hunk 1]
```diff
@@ -286,9 +290,10 @@ export const settingsLegacy = (state: SettingsState = initialState, action: Acti
 
     case 'OVERWRITE_THEN_DELETE_TOKEN_SUCCESS': {
       // where oldCurrencyCode is the sender, and tokenObj.currencyCode is the receiver (new code)
-      const receiverCode = data.tokenObj.currencyCode
-      const senderCode = data.oldCurrencyCode
-      const { tokenObj } = data
+      if (!action.data) throw new Error('Invalid action')
+      const receiverCode = action.data.tokenObj.currencyCode
+      const senderCode = action.data.oldCurrencyCode
```
Comment (kylanhurt) [inline src/modules/UI/Settings/reducer.js]: Should this be checking to make sure that `tokenObj` exists?

File: src/modules/UI/Settings/reducer.js [hunk 2]
```diff
@@ -334,15 +340,17 @@ export const settingsLegacy = (state: SettingsState = initialState, action: Acti
...
-        [currencyCode]: data
+        [currencyCode]: action.data
       }
     }
 
     case 'ADD_NEW_CUSTOM_TOKEN_SUCCESS': {
-      const { tokenObj, newCurrencyCode, settings } = data
+      if (!action.data) throw new Error('Invalid action')
+      const { tokenObj, newCurrencyCode, settings } = action.data
       const customTokens = settings.customTokens
```
Comment (kylanhurt) [inline src/modules/UI/Settings/reducer.js]: Do we need to check the existence of `settings` before we try to access the `settings.customTokens` property?

File: src/modules/UI/Settings/reducer.js (hunk 1 above)
Comment (wolverineks) [reply src/modules/UI/Settings/reducer.js]: the point of this PR is not to fix existing bugs, but to fix bugs caused by the previous action typing PR

Comment (wolverineks) [reply src/modules/UI/Settings/reducer.js]: the point of this PR is not to fix existing bugs, but to fix bugs caused by the previous action typing PR

**Pull request merged**